src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -206,4 +195,15 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
 
 @end
 
+@interface MTRDevice (Deprecated)
+
+/**
+ * Deprecated MTRDevice APIs.
+ */
++ (instancetype)deviceWithNodeID:(uint64_t)nodeID
+                deviceController:(MTRDeviceController *)deviceController
+    MTR_NEWLY_DEPRECATED("Please use deviceWithNodeID:controller:");
```

Comment:
Not 100% sure I love the MTR_NEWLY_DEPRECATED, feels like we should use the normal OS macros here..

Reply:
We can't use the normal OS macros without giving them version numbers.